viamrobotics/viam-python-sdk
# RSDK-4197: Add support for Flat Tensors in Python SDK

Use numpy arrays as the backing data struct instead of dicts for mlmodel

Test plan notes:

* EDGE FlatTensors without any tensors
    * Expected behavior: ummm I think just a empty dict

* FlatTensors with one tensor (say.. double type)
    * Expected behavior: one entry dict with same name and same structured values as `google.protobuf.internal.containers.RepeatedScalarFieldContainer`
        * Check dtype is float64

* FlatTensors with two tensors (double type, float type)
    * Expected behavior: two entry dict with same names and same structured values as `google.protobuf.internal.containers.RepeatedScalarFieldContainer`
        * Check that dtype matches double and float respectively
        * Check that the shape of the float one is 2, 2

* Check all flat tensor data types in bundles of appropriate types
